### ISSUE TITLE
Bio.Statistics.lowess breaks on some inputs

### ISSUE THREAD

**Issue by anntzer** [biopython/biopython#344]
Not sure what lowess is doing by itself in Bio.Statistics but it's nice to have a pure-Python implementation available.
In any case, it is a pity that a function supposed to smooth data fails on perfectly smooth data:

```
In [1]: lowess(arange(20), [1] * 20)
Out[1]: 
array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,
        nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])
```

(and it will also fail on weakly noisy data).

To fix this:
- the explicit matrix inversion at lines 87-89 of `lowess.py` must be replaced by a call to `numpy.linalg.lstsq`, such as `beta1, beta2 = linalg.lstsq([[A11, A12], [A12, A22]], [b1, b2])[0]`
- an explicit check for all-zeros entries must be added at the top (otherwise the algorithm gets confused by null matrices), such as `if abs(y).max() == 0: return y`.

**Comment by peterjc** [comment]
This code has not been touched other than minor style issues for some years.

I suggest we deprecate ``Bio.Statistics.lowess``, unless someone wants to put @anntzer's fix into practice?

**Comment by anntzer** [comment]
Closed by #1957.
